core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2ServiceModule.java

```diff
@@ -44,11 +36,13 @@ protected void setup(Binder binder)
         newOptionalBinder(binder, OAuth2WebUiInstalled.class);
 
         configBinder(binder).bindConfig(OAuth2Config.class);
+        configBinder(binder).bindConfig(OAuth2ClientConfig.class);
         binder.bind(OAuth2Service.class).in(Scopes.SINGLETON);
         binder.bind(OAuth2TokenHandler.class).to(OAuth2TokenExchange.class).in(Scopes.SINGLETON);
+        binder.bind(NimbusHttpClient.class).to(NimbusAirliftHttpClient.class).in(Scopes.SINGLETON);
```

Comment:
Do we need this to be injectable and available in other modules? I'd delete `NimbusHttpClient` interface and create the instance of `NimbusAirliftHttpClient` directly inside `NimbusClient`. Also, `NimbusAirliftHttpClient` could be package-private as it shouldn't be available anywhere else.